src/main/scala/workflow/DefaultOptimizer.scala

```diff
@@ -5,8 +5,9 @@ package workflow
  */
 object DefaultOptimizer extends Optimizer {
   protected val batches: Seq[Batch] =
-    Batch("DAG Optimization", FixedPoint(100), EquivalentNodeMergeRule) ::
+    Batch("DAG Optimization", FixedPoint(Int.MaxValue), EquivalentNodeMergeRule) ::
```

Comment:
Is this dangerous? Could we end-up in a (near) infinite loop if we have a really bad looking DAG?